src/api.ts

```diff
@@ -50,19 +51,22 @@ export async function synchronize(
 
   const connections = filterConnectionsAsRequired(mapToConnectionReferences(conn), params.only);
   const processes = connections.map(connection => () =>
-    withTransaction(connection, trx =>
-      runSynchronize(trx, {
-        config,
-        params,
-        connectionId: connection.id,
-        migrateFunc: t =>
-          invokeMigrationApi(t, KnexMigrationAPI.MIGRATE_LATEST, {
-            config,
-            connectionId: connection.id,
-            knexMigrationConfig: knexMigrationConfig(connection.id),
-            params: { ...params, onSuccess: params.onMigrationSuccess, onFailed: params.onMigrationFailed }
-          })
-      })
+    withTransaction(
+      connection,
+      trx =>
+        runSynchronize(trx, {
+          config,
+          params,
+          connectionId: connection.id,
+          migrateFunc: t =>
+            invokeMigrationApi(t, KnexMigrationAPI.MIGRATE_LATEST, {
+              config,
+              connectionId: connection.id,
+              knexMigrationConfig: knexMigrationConfig(connection.id),
+              params: { ...invokeParams, onSuccess: params.onMigrationSuccess, onFailed: params.onMigrationFailed }
+            })
+        }),
+      params['dry-run']
```

Comment:
Why are you passing the dry-run params separately here. The whole params object has been already passed above - shouldn't you use that?

Reply:
Understood now. Ignore this comment.